File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int j = target - nums[i];
            if(map.containsKey(j)){
                return new int[] {map.get(j),i};
            }
            map.put(nums[i], i);
        }
        return new int[] {0,0};
    }
}

File: 0001-two-sum/0001-two-sum.cpp
//#include <unordered_map>
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        //Index, nums
        unordered_map<int, int> map;
        int idx = 0;
        for(int i : nums){
            if(map.empty()){
                map[i] = idx++;
            }else{
                //target이 되는 형태가 있는 경우
                if(map.find(target - i) != map.end()){
                    return {map[target - i], idx};
                }else{
                    map[i] = idx++;
                }
            }
        }
        return {0};
    }
};